compiler/rustc_session/src/config.rs

```diff
@@ -1915,15 +1923,43 @@ fn parse_remap_path_prefix(
     matches: &getopts::Matches,
     error_format: ErrorOutputType,
 ) -> Vec<(PathBuf, PathBuf)> {
-    matches
-        .opt_strs("remap-path-prefix")
+    enum RemapType {
+        PathPrefix(String),
+        CompilationDir(String),
+    }
+    let mut positions = Vec::new();
+    positions.extend(
+        matches
+            .opt_strs_pos("remap-path-prefix")
+            .into_iter()
+            .map(|(i, s)| (i, RemapType::PathPrefix(s))),
+    );
+    positions.extend(
+        matches
+            .opt_strs_pos("debug-compilation-dir")
+            .into_iter()
+            .map(|(i, s)| (i, RemapType::CompilationDir(s))),
+    );
+    positions.sort_by(|a, b| a.0.cmp(&b.0));
```

Comment:
I assume the intent here is to make sure the relative ordering of `remap-path-prefix` and `debug-compilation-dir` options is preserved. But I wonder how much that matters? Presumably the `remap-path-prefix` option won't contain the cwd as a prefix, since avoiding that is the whole point of `debug-compilation-dir`.

Also will there ever be more than one instance of `debug-compilation-dir`? What does it mean to map the cwd multiple times?

Reply:
That's right. I don't know that it matters much, I don't expect that most people would be mixing these, but it made for an easy way to make something well-defined, rather than one command-line argument having precedence over the other.

It would be slightly simpler code to not maintain the ordering, but I considered it a net win to have something more easily explained. In this case `--debug-compilation-dir` acts exactly like an alias for `remap-path-prefix=$(pwd)=`. Anything else seems to require more explanation in the docs, and more risk of arbitrary change some edge case in the future.